spec/PushController.spec.js

```diff
@@ -866,28 +849,19 @@ describe('PushController', () => {
       installation.set('deviceType', 'ios');
       installations.push(installation);
     }
-    reconfigureServer({
+    await reconfigureServer({
       push: { adapter: pushAdapter },
       scheduledPush: true,
-    })
-      .then(() => {
-        const config = Config.get(Parse.applicationId);
-        return Parse.Object.saveAll(installations)
-          .then(() => {
-            return pushController.sendPush(payload, {}, config, auth);
-          })
-          .then(() => new Promise(resolve => setTimeout(resolve, 300)));
-      })
-      .then(() => {
-        const query = new Parse.Query('_PushStatus');
-        return query.find({ useMasterKey: true }).then(results => {
-          expect(results.length).toBe(1);
-          const pushStatus = results[0];
-          expect(pushStatus.get('status')).toBe('scheduled');
-        });
-      })
-      .then(done)
-      .catch(done.err);
+    });
+    const config = Config.get(Parse.applicationId);
+    await Parse.Object.saveAll(installations);
+    await pushController.sendPush(payload, {}, config, auth);
+    await sleep(1000);
```

Comment:
is 1000ms necessary or could we do with 300 or 500? sounds minuscule but it makes a difference when multiplying by n tests we run every month.

Reply:
These tests randomly fail and it seems to be due to timing. I haven't seen the failure since I increased the times. I believe the small delay of a second per test is less than having to re-run all of the tests again due to a failure.

Reply:
sure, let's go with 1s then